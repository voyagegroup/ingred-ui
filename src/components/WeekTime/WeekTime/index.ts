export { default } from "./WeekTime";
export type { WeekTimeProps } from "./WeekTime";
