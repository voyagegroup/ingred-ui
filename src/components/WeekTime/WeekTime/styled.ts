import styled from "styled-components";

export const Container = styled.div``;

export const WeekTimeItem = styled.span<{ active: boolean }>`
  background-color: ${({ active, theme }) =>
    // eslint-disable-next-line no-nested-ternary
    active ? theme.palette.primary.main : theme.palette.white};
  border: 1px solid
    ${({ theme, active }) =>
      // eslint-disable-next-line no-nested-ternary
      active ? theme.palette.primary.dark : theme.palette.gray.main};
  border-radius: ${({ theme }) => theme.spacing / 2}px;
  height: ${({ theme }) => theme.spacing * 3}px;
  width: ${({ theme }) => theme.spacing * 3}px;
  outline: none;
`;
