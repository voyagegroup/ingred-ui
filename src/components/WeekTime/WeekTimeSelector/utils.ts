const bin2hex = (bin: string) => {
  let out = "";
  for (let i = 0; i < bin.length; i += 4) {
    const c = bin.substr(i, 4);
    switch (c) {
      case "0000":
        out += "0";
        break;
      case "0001":
        out += "1";
        break;
      case "0010":
        out += "2";
        break;
      case "0011":
        out += "3";
        break;
      case "0100":
        out += "4";
        break;
      case "0101":
        out += "5";
        break;
      case "0110":
        out += "6";
        break;
      case "0111":
        out += "7";
        break;
      case "1000":
        out += "8";
        break;
      case "1001":
        out += "9";
        break;
      case "1010":
        out += "a";
        break;
      case "1011":
        out += "b";
        break;
      case "1100":
        out += "c";
        break;
      case "1101":
        out += "d";
        break;
      case "1110":
        out += "e";
        break;
      case "1111":
        out += "f";
        break;
      default:
        return "";
    }
  }

  return out;
};

export const convertTargetSettingToHex = (targetSetting: string[][]) => {
  let bin = "";
  for (let d = 0; d < 7; d++) {
    for (let h = 0; h < 24; h++) {
      bin += targetSetting[d][h];
    }
  }

  return bin2hex(bin);
};

/**
 * 開始位置と終了位置のそれぞれのの曜日と時間帯から、選択範囲を取得する
 */
export const getNewWeekTimeList = (
  startWeekIndex: number,
  startTimeIndex: number,
  endWeekIndex: number,
  endTimeIndex: number,
  newValue: string,
  weekTimeList: string[][],
) => {
  const newWeekTimeList = [...weekTimeList];
  const minWeekIndex = Math.min(startWeekIndex, endWeekIndex);
  const maxWeekIndex = Math.max(startWeekIndex, endWeekIndex);
  const minTimeIndex = Math.min(startTimeIndex, endTimeIndex);
  const maxTimeIndex = Math.max(startTimeIndex, endTimeIndex);

  for (let i = minWeekIndex; i <= maxWeekIndex; i++) {
    const weekTime = newWeekTimeList[i];
    const newWeekTime = weekTime.map((t, index) => {
      if (
        i >= minWeekIndex &&
        i <= maxWeekIndex &&
        index >= minTimeIndex &&
        index <= maxTimeIndex
      ) {
        return newValue; // 選択範囲内のセルの場合、newValueに更新
      }
      return t;
    });

    newWeekTimeList[i] = newWeekTime;
  }

  return newWeekTimeList;
};
