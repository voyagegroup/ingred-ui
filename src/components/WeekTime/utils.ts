export const hex2bin = (hex: string) => {
  if (!hex) return Array(7).fill(Array(24).fill("1")).join("");

  let out = "";
  for (const c of hex) {
    switch (c) {
      case "0":
        out += "0000";
        break;
      case "1":
        out += "0001";
        break;
      case "2":
        out += "0010";
        break;
      case "3":
        out += "0011";
        break;
      case "4":
        out += "0100";
        break;
      case "5":
        out += "0101";
        break;
      case "6":
        out += "0110";
        break;
      case "7":
        out += "0111";
        break;
      case "8":
        out += "1000";
        break;
      case "9":
        out += "1001";
        break;
      case "a":
        out += "1010";
        break;
      case "b":
        out += "1011";
        break;
      case "c":
        out += "1100";
        break;
      case "d":
        out += "1101";
        break;
      case "e":
        out += "1110";
        break;
      case "f":
        out += "1111";
        break;
      default:
        return "";
    }
  }

  return out;
};

/**
 * @param hex odr_targeting_week_time の16進数表記
 * @returns 週ごとの時間帯ターゲティングの設定の多重配列。[曜日][時間]の順（0番目が月曜日、0番目のn番目がn時）
 */
export const getTargetSetting = (hex?: string): string[][] => {
  if (!hex) return Array(7).fill(Array(24).fill("1"));

  const binary = hex2bin(hex);
  const times: string[][] = [];

  for (let d = 0; d < 7; d++) {
    times[d] = [];
    for (let h = 0; h < 24; h++) {
      times[d][h] = binary.charAt(d * 24 + h);
    }
  }

  return times;
};
