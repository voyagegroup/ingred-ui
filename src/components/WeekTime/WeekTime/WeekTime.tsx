import * as Styled from "./styled";
import React from "react";
import { useLocaleProps } from "../../../hooks/useLocaleProps";
import { WeekTimeElement } from "../internal/WeekTimeElement";

export type WeekTimeProps = {
  weekTime: string;
};

const WeekTime = React.forwardRef<HTMLDivElement, WeekTimeProps>(
  function WeekTime(inProps, ref) {
    const props = useLocaleProps({
      props: inProps,
      name: "WeekTime",
    });
    const { weekList, weekTime } = props;

    return (
      <Styled.Container ref={ref}>
        <WeekTimeElement
          weekTime={weekTime}
          weekList={weekList}
          WeekTimeItem={Styled.WeekTimeItem}
        />
      </Styled.Container>
    );
  },
);

export default WeekTime;
