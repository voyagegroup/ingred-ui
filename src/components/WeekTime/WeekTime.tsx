import * as Styled from "./styled";
import React, { Fragment, useMemo } from "react";
import { timeList } from "./constants";
import { getTargetSetting } from "./utils";
import { useLocaleProps } from "../../hooks/useLocaleProps";

export type WeekTimeProps = {
  weekTime: string;
};

const WeekTime: React.FC<WeekTimeProps> = (inProps) => {
  const props = useLocaleProps({
    props: inProps,
    name: "WeekTimeSelector",
  });

  const { weekList, weekTime } = props;

  const weekTimeList = useMemo(() => getTargetSetting(weekTime), [weekTime]);

  return (
    <Styled.Container>
      <Styled.WeekTimeContainer>
        <Styled.EmptyContainer />
        {timeList.map((time) => (
          <Styled.TimeContainer key={time}>{time}</Styled.TimeContainer>
        ))}
        {weekTimeList.map((time, weekIndex) => (
          // eslint-disable-next-line react/no-array-index-key
          <Fragment key={weekIndex}>
            <Styled.WeekContainer>{weekList[weekIndex]}</Styled.WeekContainer>
            {time.map((t, timeIndex) => (
              <Styled.WeekTimeItem
                // eslint-disable-next-line react/no-array-index-key
                key={`${weekIndex}-${timeIndex}`}
                active={t === "1"}
              />
            ))}
          </Fragment>
        ))}
      </Styled.WeekTimeContainer>
    </Styled.Container>
  );
};

export default WeekTime;
