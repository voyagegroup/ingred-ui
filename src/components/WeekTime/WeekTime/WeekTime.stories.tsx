import * as React from "react";
import { StoryObj } from "@storybook/react";
import WeekTime, { WeekTimeProps } from "./WeekTime";

export default {
  title: "Components/Data Display/WeekTime",
  component: WeekTime,
};

export const Example: StoryObj<WeekTimeProps> = {
  args: {
    weekTime: "ffffffffffffffffffffffffffffffffffffffffff",
  },
  render: (args) => {
    return <WeekTime weekTime={args.weekTime} />;
  },
};
