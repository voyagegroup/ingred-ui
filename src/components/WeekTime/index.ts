import WeekTime, { WeekTimeProps } from "./WeekTime";
import WeekTimeSelector, { WeekTimeSelectorProps } from "./WeekTimeSelector";

export { WeekTime, WeekTimeSelector };
export type { WeekTimeProps, WeekTimeSelectorProps };
