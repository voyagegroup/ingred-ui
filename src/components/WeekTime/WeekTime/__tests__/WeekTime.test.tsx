import * as React from "react";
import "@testing-library/jest-dom";
import { cleanup } from "@testing-library/react";
import { renderWithThemeProvider } from "../../../../utils/renderWithThemeProvider";
import WeekTime from "../WeekTime";

describe("WeekTime component testing", () => {
  afterEach(cleanup);

  test("WeekTime", () => {
    const { asFragment } = renderWithThemeProvider(
      <WeekTime weekTime="ffffffffffffffffffffffffffffffffffffffffff" />,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
