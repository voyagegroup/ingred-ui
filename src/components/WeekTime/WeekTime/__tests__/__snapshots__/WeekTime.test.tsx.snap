// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeekTime component testing WeekTime 1`] = `
<DocumentFragment>
  <div
    class="sc-bczRLJ"
  >
    <div
      class="sc-dkzDqf hGYYRO"
    >
      <div
        class="sc-hKMtZM sc-gKXOVf ijKKtp ftUcXJ"
      />
      <div
        class="sc-hKMtZM sc-jSMfEi ijKKtp hOVgha"
      >
        0
      </div>
      <div
        class="sc-hKMtZM sc-jSMfEi ijKKtp hOVgha"
      >
        1
      </div>
      <div
        class="sc-hKMtZM sc-jSMfEi ijKKtp hOVgha"
      >
        2
      </div>
      <div
        class="sc-hKMtZM sc-jSMfEi ijKKtp hOVgha"
      >
        3
      </div>
      <div
        class="sc-hKMtZM sc-jSMfEi ijKKtp hOVgha"
      >
        4
      </div>
      <div
        class="sc-hKMtZM sc-jSMfEi ijKKtp hOVgha"
      >
        5
      </div>
      <div
        class="sc-hKMtZM sc-jSMfEi ijKKtp hOVgha"
      >
        6
      </div>
      <div
        class="sc-hKMtZM sc-jSMfEi ijKKtp hOVgha"
      >
        7
      </div>
      <div
        class="sc-hKMtZM sc-jSMfEi ijKKtp hOVgha"
      >
        8
      </div>
      <div
        class="sc-hKMtZM sc-jSMfEi ijKKtp hOVgha"
      >
        9
      </div>
      <div
        class="sc-hKMtZM sc-jSMfEi ijKKtp hOVgha"
      >
        10
      </div>
      <div
        class="sc-hKMtZM sc-jSMfEi ijKKtp hOVgha"
      >
        11
      </div>
      <div
        class="sc-hKMtZM sc-jSMfEi ijKKtp hOVgha"
      >
        12
      </div>
      <div
        class="sc-hKMtZM sc-jSMfEi ijKKtp hOVgha"
      >
        13
      </div>
      <div
        class="sc-hKMtZM sc-jSMfEi ijKKtp hOVgha"
      >
        14
      </div>
      <div
        class="sc-hKMtZM sc-jSMfEi ijKKtp hOVgha"
      >
        15
      </div>
      <div
        class="sc-hKMtZM sc-jSMfEi ijKKtp hOVgha"
      >
        16
      </div>
      <div
        class="sc-hKMtZM sc-jSMfEi ijKKtp hOVgha"
      >
        17
      </div>
      <div
        class="sc-hKMtZM sc-jSMfEi ijKKtp hOVgha"
      >
        18
      </div>
      <div
        class="sc-hKMtZM sc-jSMfEi ijKKtp hOVgha"
      >
        19
      </div>
      <div
        class="sc-hKMtZM sc-jSMfEi ijKKtp hOVgha"
      >
        20
      </div>
      <div
        class="sc-hKMtZM sc-jSMfEi ijKKtp hOVgha"
      >
        21
      </div>
      <div
        class="sc-hKMtZM sc-jSMfEi ijKKtp hOVgha"
      >
        22
      </div>
      <div
        class="sc-hKMtZM sc-jSMfEi ijKKtp hOVgha"
      >
        23
      </div>
      <div
        class="sc-hKMtZM sc-eCYdqJ ijKKtp gsFxfl"
      >
        Sun
      </div>
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <div
        class="sc-hKMtZM sc-eCYdqJ ijKKtp gsFxfl"
      >
        Mon
      </div>
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <div
        class="sc-hKMtZM sc-eCYdqJ ijKKtp gsFxfl"
      >
        Tue
      </div>
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <div
        class="sc-hKMtZM sc-eCYdqJ ijKKtp gsFxfl"
      >
        Wed
      </div>
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <div
        class="sc-hKMtZM sc-eCYdqJ ijKKtp gsFxfl"
      >
        Thu
      </div>
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <div
        class="sc-hKMtZM sc-eCYdqJ ijKKtp gsFxfl"
      >
        Fri
      </div>
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <div
        class="sc-hKMtZM sc-eCYdqJ ijKKtp gsFxfl"
      >
        Sat
      </div>
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
      <span
        class="sc-gsnTZi iQtOmt"
      />
    </div>
  </div>
</DocumentFragment>
`;
